crates/red_knot_python_semantic/src/types.rs

```diff
@@ -151,6 +162,14 @@ pub(crate) fn definitions_ty<'db>(
     }
 }
 
+/// Union an iterable of declared types.
+pub(crate) fn declarations_ty<'db>(
```

Comment:
Can we make this function private or is this a common operaiton when doing type inference?

Reply:
No, I doubt we'll need it anywhere else. I can probably just inline and eliminate it; it only has one callsite. I only kept it for parallel with `bindings_ty`, but I don't think that's necessary.